Remove pin on SQLAlchemy version < 2.0

jace opened the pull request:
WTForms-SQLAlchemy with SQLAlchemy 2.0 is known to be working.

azmeuk commented:
CI is failing. Under normal circumstances, there is at least one SQLAlchemy 2 deprecation warning raised when running the tests.

```
tests/test_main.py::QuerySelectFieldTest::test_with_query_factory
  /home/eloi/dev/wtforms/wtforms-sqlalchemy/tests/test_main.py:77: RemovedIn20Warning: Deprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. To prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". Set environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message. (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)
    mapper(Test, test_table)
```

jace commented:
I'll review and push a fix tonight.

jace commented:
SQLAlchemy 1.4 is now the minimum version as the location of `declarative_base` moved from `sqlalchemy.ext.declarative` to `sqlalchemy.orm` in 1.4 and was removed from the old location in 2.0.

Since SQLAlchemy has recommended use of `inspect` since 0.8, I've switched to it in `wtforms_sqlalchemy.orm`.

`mapper.iterate_properties` does not return relationships unless mappers have been fully initialized, which appears to be delayed in 2.0. However, accessing `mapper.attrs` causes initialization, so I've switched from `mapper.iterate_properties` to `mapper.attrs.values`.

There is a `LegacyAPIWarning` emitted due to the use of `Query.get()`, which is deprecated in 2.0 but not removed. Switching to the recommended `select` construct is not taken up here.

jace commented:
Sorry for the merge commits. This will need a squash. Should I force-push one?

Also, does this need additional tests? I had to manually swap SQLAlchemy versions between test runs as it's not in the tox config. It may be worth adding (I'm not familiar with tox).

azmeuk commented:
No need to force push, there is no conflict so this will be fine.
Now the only thing needed is that I find a bit of time to spend on reviewing this.

azmeuk commented:
Thank you!
